dogstatsd-client/src/lib.rs

```diff
@@ -333,12 +351,56 @@ mod test {
             Histogram(_, _, _) => {}
             Set(_, _, _) => {}
         }
-
         // TODO: when std::mem::variant_count is in stable we can do this instead
         // assert_eq!(
         //     std::mem::variant_count::<DogStatsDActionOwned>(),
         //     std::mem::variant_count::<DogStatsDAction<String, Vec<&Tag>>>(),
         //     "DogStatsDActionOwned and DogStatsDAction should have the same number of variants,
         // did you forget to update one?", );
     }
+    use std::sync::Arc;
+    use std::thread;
+
+    #[tokio::test]
+    async fn test_thread_safety() {
```

Comment:
I'm not sure how useful this test is? It is just checking that we don't set a client until after we send. I couldn't think of a better way to unit test this.